data/kernels/filmic.cl

```diff
@@ -535,22 +535,20 @@ static inline float4 gamut_mapping(float4 Ych_final, float4 Ych_original, float4
   {
     // Now, it is still possible that one channel > display white or < display black because of saturation.
     // We have already clipped Y, so we know that any problem now is caused by c
-    pix_out = gamut_check_RGB(input_matrix, output_matrix, display_black, display_white, Ych_final);
+    return gamut_check_RGB(input_matrix, output_matrix, display_black, display_white, Ych_final);
```

Comment:
You can't return early in a branch in OpenCL, that causes issues in the vectorization. The sequence of execution needs to be independent from the data.

Reply:
Hmm ok, that can be easily changed. It's a bit surprising though, since there is data-dependent control flow anyway. The code builds just fine for me. Is there a specification or any other resource that says this is prohibited? I'd tend to think since this `if-else` structure is the last thing in the function, the compiler could figure this out. This is not exactly returning early in a branch. If I'd do

```cl
if(some_condition)
{
  // do processing...
  return function_1();
}

// some other code

return result;
```
that would be a different matter.

I can change this if it causes real issues.

Reply:
Ok, fixed.

Reply:
Thanks for the link btw.